pkg/controller/backup_configuration.go

```diff
@@ -0,0 +1,249 @@
+	util2 "github.com/appscode/stash/client/clientset/versioned/typed/stash/v1beta1/util"
+	"github.com/appscode/stash/pkg/docker"
+	"github.com/appscode/stash/pkg/util"
+	"github.com/golang/glog"
+	"k8s.io/api/batch/v1beta1"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/reference"
+)
+
+func (c *StashController) initBackupConfigurationWatcher() {
+	c.bcInformer = c.stashInformerFactory.Stash().V1beta1().BackupConfigurations().Informer()
+	c.bcQueue = queue.New("BackupConfiguration", c.MaxNumRequeues, c.NumThreads, c.runBackupConfigurationInjector)
+	c.bcInformer.AddEventHandler(queue.NewObservableUpdateHandler(c.bcQueue.GetQueue(), apis.EnableStatusSubresource))
+	c.bcLister = c.stashInformerFactory.Stash().V1beta1().BackupConfigurations().Lister()
+}
+
+// syncToStdout is the business logic of the controller. In this controller it simply prints
+// information about the deployment to stdout. In case an error happened, it has to simply return the error.
+// The retry logic should not be part of the business logic.
+func (c *StashController) runBackupConfigurationInjector(key string) error {
+	obj, exists, err := c.bcInformer.GetIndexer().GetByKey(key)
+	if err != nil {
+		glog.Errorf("Fetching object with key %s from store failed with %v", key, err)
+		return err
+	}
+	if !exists {
+		glog.Errorf("BackupConfiguration %s does not exit anymore\n", key)
+
+	} else {
+		backupconfiguration := obj.(*api_v1beta1.BackupConfiguration)
+		glog.Infof("Sync/Add/Update for BackupConfiguration %s", backupconfiguration.GetName())
+		if backupconfiguration.DeletionTimestamp != nil {
+			if core_util.HasFinalizer(backupconfiguration.ObjectMeta, api_v1beta1.StashKey) {
+				namespace, name, err := cache.SplitMetaNamespaceKey(key)
+				if err != nil {
+					return err
+				}
+				if err = c.EnsureV1beta1SidecarDeleted(backupconfiguration); err != nil {
+					return err
+				}
+				if err = c.EnsureCronJobDeleted(namespace, name); err != nil {
+					return err
+				}
+				// Remove finalizer
+				_, _, err = util2.PatchBackupConfiguration(c.stashClient.StashV1beta1(), backupconfiguration, func(in *api_v1beta1.BackupConfiguration) *api_v1beta1.BackupConfiguration {
+					in.ObjectMeta = core_util.RemoveFinalizer(in.ObjectMeta, api_v1beta1.StashKey)
+					return in
+
+				})
+				if err != nil {
+					return err
+				}
+			}
+		} else {
+			//Add finalizer
+			_, _, err := util2.PatchBackupConfiguration(c.stashClient.StashV1beta1(), backupconfiguration, func(in *api_v1beta1.BackupConfiguration) *api_v1beta1.BackupConfiguration {
+				in.ObjectMeta = core_util.AddFinalizer(in.ObjectMeta, api_v1beta1.StashKey)
+
+				return in
+			})
+			if err != nil {
+				return err
+			}
+			if backupconfiguration.Spec.Target != nil {
+				if backupconfiguration.Spec.Target.Ref != nil {
+					kind := backupconfiguration.Spec.Target.Ref.Kind
```

Comment:
Do we need `kind` variable here? You can just send `backupconfiguration.Spec.Target.Ref.Kind` as parameter.

Reply:
fixed